io/bufio.ts

```diff
@@ -15,23 +15,36 @@ const MAX_CONSECUTIVE_EMPTY_READS = 100;
 const CR = charCode("\r");
 const LF = charCode("\n");
 
-export type BufState =
-  | null
-  | "EOF"
-  | "BufferFull"
-  | "ShortWrite"
-  | "NoProgress"
-  | Error;
+export class BufferFullError extends Error {
+  constructor(readonly partial: Uint8Array) {
+    super("Buffer full");
+  }
+}
+
+export class UnexpectedEOFError extends Error {
+  constructor() {
+    super("Unexpected EOF");
+  }
+}
+
+export const EOF: unique symbol = Symbol("EOF");
+export type EOF = typeof EOF;
+
+/** Result type returned by of BufReader.readLine(). */
+export interface ReadLineResult {
+  line: Uint8Array;
+  more: boolean;
+}
 
 /** BufReader implements buffering for a Reader object. */
 export class BufReader implements Reader {
   private buf: Uint8Array;
   private rd: Reader; // Reader provided by caller.
   private r = 0; // buf read position.
   private w = 0; // buf write position.
-  private lastByte: number;
-  private lastCharSize: number;
-  private err: BufState;
+  private eof = false;
+  // private lastByte: number;
+  // private lastCharSize: number;
```

Comment:
Are these intentionally unused? If so remove.

Reply:
Intentional.
You should answer the question why this commented-out code was there in the first place, it seems that they are related to some methods (`unreadByte()`) that aren't ported over (yet).
I just made it a bit more consistent (commenting out assignments to related private properties that are never read).